api/src/main/java/com/datastrato/gravitino/authorization/SecurableObjects.java

```diff
@@ -29,7 +33,41 @@ public static SecurableObject of(String... names) {
       throw new IllegalArgumentException("Cannot create a securable object with no names");
     }
 
+    if (type == null) {
+      throw new IllegalArgumentException("Cannot create a securable object with no type");
+    }
+
+    if (names.length > 3) {
+      throw new IllegalArgumentException(
+          "Cannot create a securable object with the name length which is greater than 3");
+    }
+
+    List<SecurableObjectType> typeInDifferentLevels =
+        Lists.newArrayList(SecurableObjectType.CATALOG, SecurableObjectType.SCHEMA);
+
+    if (names.length == 1 && type != SecurableObjectType.CATALOG) {
+      throw new IllegalArgumentException(
+          "If the length of names is 1, it must be the CATALOG type");
+    }
+
+    if (names.length == 2 && type != SecurableObjectType.SCHEMA) {
+      throw new IllegalArgumentException("If the length of names is 2, it must be the SCHEMA type");
+    }
+
+    if (names.length == 3
+        && type != SecurableObjectType.FILESET
+        && type != SecurableObjectType.TABLE
+        && type != SecurableObjectType.TOPIC) {
+      throw new IllegalArgumentException(
+          "If the length of names is 3, it must be FILESET, TABLE or TOPIC");
+    }
+
+    if (names.length > 2) {
+      typeInDifferentLevels.add(type);
+    }
```

Comment:
What's the meaning of these codes?

Reply:
I refactor these code to make it more clear.